### ISSUE TITLE
Intentional "function result is not used"

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1083]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | U2.0.9 |
| Language level | -                      |

I don't know what we can do here. There some cases where we have some intentional call of functions (in my case methods) that have return and we don't want to use that. 

```php
class ExampleClass
{
    public function chainMethod()
    {
        return $this;
    }
}

$exampleInstance = new ExampleClass;
$exampleInstance->chainMethod();
```

The example above is a chain method that will throw this message. This case is easy to fix by just check if the return is `$this` or same instance of class. But the problem will happen if it returns a more complex value:

```php
class ExampleClass
{
    public function youCouldUseThisReturnIfYouNeed()
    {
        $createSomething = new Something;
        $this->addSomething($createSomething);
        
        return $createSomething;
    }
}

$exampleInstance = new ExampleClass;
$exampleInstance->youCouldUseThisReturnIfYouNeed();
```

For now I am disabling this inspection, but I think that it could be limited to native PHP functions that we can have sure that it return value must be usable or not.

What do you think?

**Comment by kalessil** [comment]
Hi, has been solved already: https://github.com/kalessil/phpinspectionsea/commit/78430c110adf42da2eb3b2de92f8ab839287e153
